tests/aws-cpp-sdk-cognitoidentityprovider-integration-tests/OperationTest.cpp

```diff
@@ -0,0 +1,379 @@
+
+protected:
+
+    const Aws::String m_clientName{"testClient"};
+    Aws::String m_userPoolId;
+    Aws::String m_clientId;
+    const Aws::String m_username{"testUser"};
+    Aws::String m_clientSecret;
+
+    void SetUp()
+    {
+        Aws::Client::ClientConfiguration config;
+        config.region = AWS_TEST_REGION;
+
+        client = Aws::MakeShared<Aws::CognitoIdentityProvider::CognitoIdentityProviderClient>(ALLOCATION_TAG, config);
+
+        //make user pool
+        m_userPoolId = createUserPool("TestPool");
+
+        ASSERT_TRUE(!m_userPoolId.empty());
+
+        //make client
+        if(!m_userPoolId.empty())
+        {
+            auto ret = createPoolClient(m_userPoolId, m_clientName);
+            m_clientId = ret.first;
+            m_clientSecret = ret.second;
+        }
+
+        ASSERT_TRUE(!m_clientId.empty());
+        ASSERT_TRUE(!m_clientSecret.empty());
+
+        //create user
+        if(!m_userPoolId.empty() && !m_clientId.empty())
+        {
+            ASSERT_TRUE(createUser(m_username, "Password123!", m_userPoolId));
+        }
+
+    }
+
+    void TearDown()
+    {
+
+        //delete user
+        if(!m_userPoolId.empty())
+        {
+            ASSERT_TRUE(deleteUser(m_username, m_userPoolId));
+        }
+
+        // delete client
+        if(!m_userPoolId.empty() && !m_clientId.empty())
+        {
+            ASSERT_TRUE(deletePoolClient(m_userPoolId, m_clientId));
+        }
+
+        // delete user pool
+        if(!m_userPoolId.empty())
+        {
+            ASSERT_TRUE(deleteUserPool(m_userPoolId));
+        }
+
+
+        if (::testing::Test::HasFailure())
+        {
+            AWS_LOGSTREAM_DEBUG(ALLOCATION_TAG,  "Test traces: " << testTrace);
+        }
+        testTrace.erase();
+    }
+
+    private:
+    Aws::String createUserPool(Aws::String poolName)
+    {
+        if(poolName.empty())
+        {
+            return {};
+        }
+        Aws::CognitoIdentityProvider::Model::CreateUserPoolRequest userPoolRequest;
+        userPoolRequest.SetPoolName(poolName);
+
+        CreateUserPoolOutcome userPoolOutcome = client->CreateUserPool(userPoolRequest);
+        if (!userPoolOutcome.IsSuccess()) {
+            return {};
+        }
+
+        auto poolId = userPoolOutcome.GetResult().GetUserPool().GetId();
+        AWS_LOGSTREAM_DEBUG(ALLOCATION_TAG, "Created User Pool with ID: " << poolId << " for pool name="<< poolName );
+        return poolId;
+    }
+
+    bool deleteUserPool(const Aws::String& poolId)
+    {
+        if (poolId.empty())
+        {
+            return false;
+        }
+        Aws::CognitoIdentityProvider::Model::DeleteUserPoolRequest deletePoolRequest;
+        deletePoolRequest.SetUserPoolId(poolId);
+
+        auto deletePoolOutcome = client->DeleteUserPool(deletePoolRequest);
+        if (!deletePoolOutcome.IsSuccess()) {
+            AWS_LOGSTREAM_ERROR(ALLOCATION_TAG, "Failed to delete User Pool: " << deletePoolOutcome.GetError().GetMessage());
```

Comment:
yeah last comment ill leave on this we shoudlnt be logging errors, this is a test therefore we want to assert/expecting